python/tank/platform/qt/event_emitter.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from . import QtCore
+
+class EventEmitter(QtCore.QObject):
+    """
+    A container object for event signals.
+
+    :signal event(object): A generic event signal housing a Python object.
+        Typically this object will be an :class:`sgtk.platform.EngineEvent`
+        object.
+    """
+    event = QtCore.Signal(object)
```

Comment:
Do we have to rely on QT's signalling mechanism for this? What about lifting it out of `platform.qt` and doing a non-QT implementation, based on callbacks? We have had issues in the past with signals and deadlocks - my gut feeling tells me that unless there are obvious advantages with a signal/slot approach, we should avoid it. It also creates a dependency on QT (thinking about future javascript based environments).